src/main/java/frc/robot/scoreStates/ShootState.java

```diff
@@ -44,20 +60,49 @@ public Command driveToClosestOptimalPoint() {
                     return AutoBuilder.pathfindToPose(
                             new Pose2d(
                                     optimalTranslation,
-                                    Utils.calculateOptimalRotation(
+                                    Utils.calcRotationToTranslation(
                                             optimalTranslation, speakerPose)),
                             Constants.AUTO_CONSTRAINTS);
                 },
                 Set.of(SwerveDrive.getInstance()));
     }
 
     @Override
-    public Command stateInitialize() {
-        return null;
+    public Command initializeCommand() {
+        return Commands.none();
+    }
+
+    @Override
+    public Command initializeSubsystem() {
+        return Commands.parallel(
+                shooter.setVelocity(
+                        () ->
+                                Units.RotationsPerSecond.of(
+                                                ShooterConstants.interpolationMap.get(
+                                                                Utils.getDistanceFromPoint(
+                                                                        speakerPose,
+                                                                        PoseEstimation.getInstance()
+                                                                                .getEstimatedPose()))
+                                                        .value)
+                                        .mutableCopy()),
+                hood.setAngle(
+                        () ->
+                                Units.Degrees.of(
+                                                Math.atan(
+                                                        (Constants.SPEAKER_TARGET_POSE.getY()
+                                                                        - ShooterConstants
+                                                                                .SHOOTER_HEIGHT)
```

Comment:
If you have a calculation between two constants, just save the result in a constant. You can delete `SHOOTER_HEIGHT` on the way, it's only used here.

Reply:
f67fb1f8f228875881f49d283b8597694cd8a3de